src/CardProxy.php

```diff
@@ -10,104 +15,70 @@
  */
 class CardProxy extends Base
 {
-    protected $customerUniqueId;
-    protected $cardNumber;
-    protected $expiry;
-    protected $cardHolderName;
-    protected $paymentAmount;
+    use HasPaymentAmount;
+    use HasCarbonDates;
+
+    protected string $customerUniqueId;
+    protected string $cardNumber;
+    protected Carbon $expiry;
+    protected string $cardHolderName;
 
-    protected function getEndpoint()
+    protected function getEndpoint(): string
     {
         return 'cardproxies';
     }
 
     /**
      * @return mixed
      */
-    public function getCustomerUniqueId()
+    public function getCustomerUniqueId(): string
     {
         return $this->customerUniqueId;
     }
 
-    /**
-     * @return CardProxy
-     */
-    public function setCustomerUniqueId(mixed $customerUniqueId)
+    public function setCustomerUniqueId(string $customerUniqueId): self
     {
         $this->customerUniqueId = $customerUniqueId;
 
         return $this;
     }
 
-    /**
-     * @return mixed
-     */
-    public function getCardNumber()
+    public function getCardNumber(): string
     {
         return $this->cardNumber;
     }
 
-    /**
-     * @return CardProxy
-     */
-    public function setCardNumber(mixed $cardNumber)
+    public function setCardNumber(string $cardNumber): self
     {
         $this->cardNumber = $cardNumber;
 
         return $this;
     }
 
-    /**
-     * @return mixed
-     */
-    public function getExpiry()
+    public function getExpiry(): string
     {
-        return $this->expiry;
+        return $this->castCarbonToString($this->expiry, 'm/y');
     }
 
     /**
-     * @return CardProxy
+     * @throws SchoolEasyPayException
      */
-    public function setExpiry(mixed $expiry)
+    public function setExpiry(string $expiry): self
     {
-        $this->expiry = $expiry;
-
+        $dateStringWithDay = '01/' . $expiry;
```

Comment:
What is this doing?

Reply:
Expiry is provided in the format dd/yy.  To turn it into a carbon date object, a day of the month (in this case first of the month) is prepended to the string.  Then on the get, the carbon date is converted back to a string with the format 'dd/yy'.